src/widgets/NestedStringListArrayWidget.tsx

```diff
@@ -0,0 +1,791 @@
+                            constraints,
+                            availableSelection: allValues
+                          })
+                        }
+                      })
+                    })
+
+                    if (!bulkSelectionTriggerRef.current) return
+                    bulkSelectionTriggerRef.current.click()
+                  }}
+                >
+                  Select all
+                </ActionButton>
+              )}
+              {groupIntersectsSelection(allValues, selection[name]) ? (
+                <ActionButton
+                  type='button'
+                  aria-label={`Clear all ${label}`}
+                  onClick={ev => {
+                    ev.stopPropagation()
+                    flushSync(() => {
+                      setSelection(prevState => {
+                        return { ...prevState, [name]: [] }
+                      })
+                    })
+                    if (!bulkSelectionTriggerRef.current) return
+                    bulkSelectionTriggerRef.current.click()
+                  }}
+                >
+                  Clear all
+                </ActionButton>
+              ) : null}
+            </Actions>
+          </div>
+        </WidgetActionsWrapper>
+        <WidgetTooltip
+          helpText={help || null}
+          triggerAriaLabel={`Get help for ${label}`}
+        />
+      </WidgetHeader>
+      <ReservedSpace data-stylizable='widget string-listarray reserved-error-space'>
+        {!bypassed && required && !selection[name]?.length ? (
+          <Error>At least one selection must be made</Error>
+        ) : null}
+      </ReservedSpace>
+      <Fieldset name={name} ref={fieldSetRef} disabled={fieldsetDisabled}>
+        <Legend>{label}</Legend>
+
+        <GroupRenderer
+          name={name}
+          allGroups={allGroups}
+          groups={groups}
+          selection={selection}
+          constraints={constraints}
+          accordionOptions={accordionOptions}
+          renderActiveSelectionsCount={renderActiveSelectionsCount}
+          deep={0}
+          onActionButtonClick={onActionButtonClick}
+          setSelection={setSelection}
+        />
+      </Fieldset>
+    </Widget>
+  )
+}
+
+interface GroupRendererProps {
+  groups: StringListArrayWidgetConfiguration['details']['groups']
+  name: string
+  allGroups: StringListArrayWidgetGroupDetails[]
+  selection: Record<string, string[]>
+  constraints?: string[]
+  accordionOptions: StringListArrayWidgetDetails['accordionOptions']
+  renderActiveSelectionsCount?: boolean
+  deep: number
+  setSelection: (value: React.SetStateAction<Record<string, string[]>>) => void
+  onActionButtonClick: VoidFunction
+}
+const GroupRenderer = ({
+  groups,
+  allGroups,
+  name,
+  selection,
+  constraints,
+  accordionOptions,
+  deep,
+  renderActiveSelectionsCount,
+  setSelection,
+  onActionButtonClick
+}: GroupRendererProps) => {
+  return (
+    <>
+      {deep <= 1 &&
```

Comment:
let's see if we can avoid using `deep`

Reply:
I have not found a way around using such a property to avoid infinite nesting. I could extract it to a function, but in essence it will be the same. As you prefer